src/OpenSearch/Traits/DeprecatedPropertyTrait.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace OpenSearch\Traits;
+
+/**
+ * Provides a standard way to announce deprecated properties.
+ */
+trait DeprecatedPropertyTrait
+{
+
+    /**
+     * Checks for access to the deprecated endpoint property.
+     */
+    public function __get(string $name): mixed
+    {
+        if ($name === 'endpoints') {
+            @trigger_error('The $endpoints property is deprecated in 2.3.2 and will be removed in 3.0.0. Use $endpointFactory instead.', E_USER_DEPRECATED);
```

Comment:
here, the magic method gets called because you access the property from outside, requiring public visibility. But this is **not** what the BC layer is about. In the existing codebase, public access to `endpoints` does not work, so there is no need to add support for it in a deprecated way.

The relevant thing that you need to deprecate is accessing the protected property from a child class. And in such case, the magic method will not be called as the property itself is accessible.

Reply:
See https://3v4l.org/21VGW for the example calling this in protected context